server/src/main/java/de/uftos/services/StudentGroupService.java

```diff
@@ -145,11 +145,10 @@ public StudentGroupResponseDto update(String id, StudentGroupRequestDto groupReq
 
     List<Grade> grades =
         this.gradeRepository.findAllById(groupRequest.gradeIds());
-    List<Grade> allGrades = this.gradeRepository.findAll();
-    for (Grade grade : allGrades) {
-      grade.getStudentGroups().remove(group);
-    }
```

Comment:
Why is this loop thrown out?

Reply:
Because I replaced it with a more efficient version down below (I forgot a delete statement though. I will fix it)